### PR TITLE
UIP-47: Simplify narrative installation

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Removing antiquated `setup.py` and adding the `src/` directory to the `PYTHONPATH` env var to simplify narrative installation and use.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/UIP-47
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and checking that the backend works, or spinning up a docker image and checking that the backend works. Alternatively, just run the tests.

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook
- [x] (Python) I have run Ruff `format` and `check` on changed Python code manually or with a git precommit hook
- [ ] Any dependent changes have been merged and published in downstream modules

# Updating Version and Release Notes (if applicable)

N/A

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: scripts/build_narrative_container.sh [hunk 1]
```diff
@@ -3,12 +3,11 @@
 VER_OUTFILE="deployment/ui-common/narrative_version"
 DS=$( date +%Y%m%d%H%M )
 
-# This is the name for now, as this is what the Lua provisioner looks for to fire up a Narrative.
 NAR_NAME="kbase/narrative"
 NAR_VER_NAME="kbase/narrative_version"  # Image for serving up the narrative version
 
 # Get the current branch, so that we can tag images to branch
-BRANCH=${TRAVIS_BRANCH:-`git symbolic-ref --short HEAD`}
+BRANCH=${GIT_BRANCH:-`git symbolic-ref --short HEAD`}
```
Comment (ialarmedalien) [inline scripts/build_narrative_container.sh:10]: `TRAVIS_BRANCH` was annoying me

File: scripts/install_narrative.sh [hunk 2]
```diff
@@ -46,20 +46,20 @@ done
...
-    cd $NARRATIVE_ROOT_DIR
     npm install 2>&1 | tee -a ${logfile}
     npm run install-npm
 
     # Install Narrative requirements
     # ------------------------------
     log "Installing biokbase requirements from src/requirements.txt"
-    cd "$NARRATIVE_ROOT_DIR/src"
-    cat requirements.txt | sed -e '/^\s*#.*$/d' -e '/^\s*$/d' | xargs -n 1 pip --no-cache-dir install 2>&1 | tee -a "${logfile}"
+    cat $NARRATIVE_ROOT_DIR/src/requirements.txt | sed -e '/^\s*#.*$/d' -e '/^\s*$/d' | xargs -n 1 pip --no-cache-dir install 2>&1 | tee -a "${logfile}"
```
Comment (ialarmedalien) [inline scripts/install_narrative.sh:62]: this command is what the `install_docker_narrative.sh` script uses

File: scripts/install_narrative.sh [hunk 3]
```diff
@@ -68,27 +68,18 @@ then
...
 
-# Install Narrative code
-# ----------------------
-log "Installing biokbase modules"
-cd "$NARRATIVE_ROOT_DIR/src"
-log "Running local 'setup.py'"
-python setup.py install 2>&1 | tee -a "${logfile}"
-log "Done installing biokbase."
-cd "$NARRATIVE_ROOT_DIR"
-
```
Comment (ialarmedalien) [inline scripts/install_narrative.sh:86]: Begone foul setup.py!

File: scripts/install_narrative.sh [hunk 4]
```diff
@@ -110,7 +101,6 @@ then
     cd nbextensions
     sh install.sh
     cd ../..
-    jupyter nbextension enable --py --sys-prefix widgetsnbextension
```
Comment (ialarmedalien) [inline scripts/install_narrative.sh:113]: this line is in the install.sh script

File: scripts/install_narrative_docker.sh [hunk 5]
```diff
@@ -29,20 +29,10 @@ cat $NARRATIVE_ROOT_DIR/src/requirements.txt | sed -e '/^\s*#.*$/d' -e '/^\s*$/d
 console "Installing dev requirements from src/requirements.txt"
 cat $NARRATIVE_ROOT_DIR/src/requirements-dev.txt | sed -e '/^\s*#.*$/d' -e '/^\s*$/d' | xargs -n 1 pip install
 
-# Install Narrative code
-# ----------------------
-console "Installing biokbase modules"
-cd $NARRATIVE_ROOT_DIR/src
-console "Running local 'setup.py'"
-${PYTHON} setup.py install
-console "Done installing biokbase."
```
Comment (ialarmedalien) [inline scripts/install_narrative_docker.sh:38]: Begone again foul setup.py!

File: scripts/run_headless.tmpl [hunk 6]
```diff
@@ -1,20 +0,0 @@
...
-CFGDIR=$JUPYTER_CONFIG_DIR/static/kbase
-
-if [ -n "$ENVIRON" ]
-then
-    sed >/tmp/config.json "s/\\"config\\": \\".*\\"/\\"config\\": \\"$ENVIRON\\"/" $NARRATIVE_DIR/src/config.json && cp /tmp/config.json $NARRATIVE_DIR/src/config.json
-fi
-
-cp $NARRATIVE_DIR/src/config.json $JUPYTER_CONFIG_DIR/static/kbase/config/
-
-python /kb/dev_container/narrative/src/biokbase/narrative/exporter/run_narrative.py "$@"
```
Comment (ialarmedalien) [inline scripts/run_headless.tmpl:20]: this script does not exist. I meant to delete this file in the last PR but must have forgotten

File: scripts/start_docker_narrative.tmpl [hunk 7]
```diff
@@ -6,6 +6,7 @@ export JUPYTER_RUNTIME_DIR=/tmp/jupyter_runtime
 export JUPYTER_DATA_DIR=/tmp/jupyter_data
 export JUPYTER_PATH=$NARRATIVE_DIR/kbase-extension
 export IPYTHONDIR=$NARRATIVE_DIR/kbase-extension/ipython
+export PYTHONPATH=$NARRATIVE_DIR/src/:$PYTHONPATH
```
Comment (ialarmedalien) [inline scripts/start_docker_narrative.tmpl:9]: cool new `PYTHONPATH`

File: scripts/run_backend_tests.sh [hunk 8]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 export NARRATIVE_DIR=$(pwd)
-
+export PYTHONPATH=$NARRATIVE_DIR/src:$PYTHONPATH
```
Comment (ialarmedalien) [inline scripts/run_backend_tests.sh:3]: amend the `PYTHONPATH`

**Review by briehl — APPROVED** [review]

**Pull request merged**